src/middlewared/middlewared/event.py

```diff
@@ -2,18 +2,23 @@
 import contextlib
 import json
 import threading
+from typing import Iterable
 
+from middlewared.role import RoleManager
 from middlewared.schema import Any, clean_and_validate_arg, ValidationErrors
 from middlewared.settings import conf
 
 
 class Events:
-    def __init__(self, role_manager):
+    _events: dict[str, dict[str]]
```

Comment:
Why move them to class variables ? We can specify their types in `__init__` as well ?